test/test_internet.py

```diff
@@ -9,22 +9,61 @@
 import pytest
 import warnings
 from urllib.error import URLError
+from urllib.request import Request
 
 import torchvision.datasets.utils as utils
 from common_utils import get_tmp_dir
 
 
-class TestDatasetUtils:
+@pytest.fixture
```

Comment:
do we need it to be a fixture so that we can use `mocker`?

Reply:
Yes. We could also define a regular function and pass `mocker` though. No strong opinion here.